src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenUsingStatementTests.cs

```diff
@@ -1011,20 +1007,153 @@ .maxstack  1
   IL_0000:  newobj     ""Program.MyManagedClass..ctor()""
   IL_0005:  stloc.0
   .try
+  {
+    IL_0006:  ldloc.0
+    IL_0007:  callvirt   ""void Program.MyManagedClass.Use()""
+    IL_000c:  leave.s    IL_0018
+  }
+  finally
+  {
+    IL_000e:  ldloc.0
+    IL_000f:  brfalse.s  IL_0017
+    IL_0011:  ldloc.0
+    IL_0012:  callvirt   ""void Program.MyManagedClass.Dispose()""
+    IL_0017:  endfinally
+  }
+  IL_0018:  ret
+}");
+        }
+
+        [Fact]
+        public void UsingPatternTest()
+        {
+            var source = @"
+class C1
 {
-  IL_0006:  ldloc.0
-  IL_0007:  callvirt   ""void Program.MyManagedClass.Use()""
-  IL_000c:  leave.s    IL_0018
+    public C1() { }
+    public void Dispose() { }
+}
+
+class C2
+{
+    static void Main()
+    {
+        using (C1 c = new C1())
+        {
+        }
+    }
+}";
+            CompileAndVerify(source).VerifyIL("C2.Main()", @"
+{
+  // Code size       19 (0x13)
+  .maxstack  1
+  .locals init (C1 V_0) //c
+  IL_0000:  newobj     ""C1..ctor()""
+  IL_0005:  stloc.0
+  .try
+  {
+    IL_0006:  leave.s    IL_0012
+  }
+  finally
+  {
+    IL_0008:  ldloc.0
+    IL_0009:  brfalse.s  IL_0011
+    IL_000b:  ldloc.0
+    IL_000c:  callvirt   ""void C1.Dispose()""
```

Comment:
This seems weird to me. I could maybe understand a struct constrained callvirt, but this is a regular class. I don't understand why this isn't a `call`. Can you look at `EmitCallExpression` in the debugger for this test and see why it's choosing to use `CallKind.CallVirt` instead of `CallKind.Call`?

Reply:
Actually, I think I understand this. I bet that we don't exactly know that the local is a temp for the object creation expression, so we're doing a null test. It would be good to confirm that `CanUseCallOnRefTypeReceiver` is returning `false` here.

Reply:
I'd spoken briefly with Jared about this earlier, who said that `callvirt` was the correct option. A bit confused as to which I should stick with.

Reply:
Yeah, if there's any work to do here it's probably not part of this feature. I think we could get cleverer with some codegen, but let's not do that now.